libnetwork/network.go

```diff
@@ -1979,39 +1981,33 @@ func (n *Network) ResolveName(ctx context.Context, req string, ipType int) ([]ne
 	// TODO(aker): release the lock earlier
 	defer c.mu.Unlock()
 	sr, ok := c.svcRecords[networkID]
-
 	if !ok {
 		return nil, false
 	}
 
 	req = strings.TrimSuffix(req, ".")
 	req = strings.ToLower(req)
-	ipSet, ok := sr.svcMap.Get(req)
 
+	ipSet, ok4 := sr.svcMap.Get(req)
+	ipSet6, ok6 := sr.svcIPv6Map.Get(req)
+	if !ok4 && !ok6 {
+		// No result for v4 or v6, the name doesn't exist.
+		return nil, false
+	}
 	if ipType == types.IPv6 {
-		// If the name resolved to v4 addresses then it's a valid name in the docker
-		// network domain. If there is no v6 address, because the network or the
-		// container is not v6 enabled, set ipv6Miss to stop the DNS query from going to
-		// external nameservers.
-		ipv6Miss = ok
-		ipSet, ok = sr.svcIPv6Map.Get(req)
-	}
-
-	if ok && len(ipSet) > 0 {
-		// this map is to avoid IP duplicates, this can happen during a transition period where 2 services are using the same IP
-		noDup := make(map[string]bool)
-		var ipLocal []net.IP
-		for _, ip := range ipSet {
-			if _, dup := noDup[ip.ip]; !dup {
-				noDup[ip.ip] = true
-				ipLocal = append(ipLocal, net.ParseIP(ip.ip))
-			}
-		}
-		// TODO(robmry) - the bool return is 'ipv6Miss', it should be 'false' here.
-		return ipLocal, ok
+		ipSet = ipSet6
 	}
 
-	return nil, ipv6Miss
+	// this map is to avoid IP duplicates, this can happen during a transition period where 2 services are using the same IP
+	noDup := make(map[string]bool)
```

Comment:
Not relevant for this PR, but this could uses a `map[string]struct{}` to avoid allocations.